pkg/frontend/openshiftcluster_preflightvalidation_test.go

```diff
@@ -158,14 +158,14 @@ func TestPreflightValidation(t *testing.T) {
 				t.Fatal(err)
 			}
 
-			f, err := NewFrontend(ctx, ti.audit, ti.log, ti.env, ti.asyncOperationsDatabase, ti.clusterManagerDatabase, ti.openShiftClustersDatabase, ti.subscriptionsDatabase, ti.openShiftVersionsDatabase, api.APIs, &noop.Noop{}, &noop.Noop{}, nil, nil, nil, nil, nil)
+			f, err := NewFrontend(ctx, ti.audit, ti.log, ti.env, ti.asyncOperationsDatabase, ti.clusterManagerDatabase, ti.openShiftClustersDatabase, ti.subscriptionsDatabase, ti.openShiftVersionsDatabase, ti.platformWorkloadIdentityRoleSetsDatabase, api.APIs, &noop.Noop{}, &noop.Noop{}, nil, nil, nil, nil, nil)
 			if err != nil {
 				t.Fatal(err)
 			}
 			oc := tt.preflightRequest()
 
 			go f.Run(ctx, nil, nil)
-			f.mu.Lock()
+			f.ocpVersionsMu.Lock()
```

Comment:
This might be ok - but whenever I see changes to concurrency I kinda freak out and get antsy :p is it required to redo how we perform frontend mutex locking for this change, or is it a premature optimization outside the acceptance criteria? These changes feel like something, given the scope of other changes, that might be better in their own PR so it is more clear exactly what changes are needed for this concurrency updates, and what the benefits are, if at all possible.

Reply:
The reason for the rename of this mutex lock and the addition of a second one for the `PlatformWorkloadIdentityRoleSets` is that it logically doesn't make sense for one goroutine to be unable to edit the `PlatformWorkloadIdentityRoleSets` because another is editing the `OpenShiftVersions` in Cosmos DB.

I could use one mutex for this PR and add the second in a follow-up if that would be better, but we have to modify concurrency stuff no matter what in this PR because we have to use _a mutex lock_ around accesses to the `PlatformWorkloadIdentityRoleSets` - it's just a matter of whether it's the same old one or the new one I added.

Reply:
correct - I'm just trying to wrap my head around concurrent writes to different tables. I understand that from a Cosmos perspective, we should be able to write to both containers at the same time, but my question is, can the RP handle that? Would it be safer code-wise to keep all locks as the same thing, even if it is inefficient, so we don't risk a future change mistakingly using the wrong lock, because the RP doesn't write enough to Cosmos for us to care? ... Maybe I'm over-thinking this. :sweat_smile:

Reply:
Oh, I see what you're saying. I agree that functionally, for us, one mutex lock for both containers would be fine since we edit both containers very rarely and they each contain a small amount of data and hence each won't one doesn't take long to update.